### PR TITLE
fix(alert): remove border-right on ios stacked buttons

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

When iOS alert has 3 or more buttons, those are rendered vertically. All but the last button will have a right border. There shouldn't be a right border when the buttons are stacked.

Group 2(1)

## What is the new behavior?

- Vertical buttons don't have a right border.
- Separated the test in order to also test dark theme.

Screenshot 2024-01-12 at 1 00 54 PM
Screenshot 2024-01-12 at 12 58 22 PM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

N/A

### CODE REVIEW

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png
Comment (brandyscarney) [inline core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png:1]: It looks like there is still a right border on the last button in the new screenshot.

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png
Comment (brandyscarney) [inline core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png:1]: Can we make the delete button red "destructive" again?

Comment (thetaPC) [reply core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png:1]: https://github.com/ionic-team/ionic-framework/pull/28821/commits/1723e2dccbcfc76bbf66dfa553dc82859f14d866

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png
Comment (thetaPC) [reply core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png:1]: Good catch! https://github.com/ionic-team/ionic-framework/pull/28821#pullrequestreview-1824441664

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png
Comment (brandyscarney) [reply core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png:1]: Wait, sorry. I didn't realize this was custom CSS. I thought it was the default for both modes. We should remove that then.

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png
Comment (thetaPC) [reply core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-ios-rtl-Mobile-Chrome-linux.png:1]: https://github.com/ionic-team/ionic-framework/pull/28821/commits/d3c49cd552055a0a506cfabdfa99ec7697ea8366

**Review by sean-perkins — APPROVED** [review]

**Review by brandyscarney — APPROVED** [review]

File: core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png
Comment (thetaPC) [reply core/src/components/alert/test/basic/alert.e2e.ts-snapshots/alert-multipleButtons-md-ltr-Mobile-Chrome-linux.png:1]: Done https://github.com/ionic-team/ionic-framework/pull/28821/commits/b43e76d6c0afa77c91ac94c228a6d7c8b80dc9da

**Pull request merged**